pkg/kubelet/cm/pod_container_manager_linux_test.go

```diff
@@ -126,3 +135,160 @@ func TestIsCgroupPod(t *testing.T) {
+				cgroupManager: NewCgroupManager(nil, "cgroupfs"),
+			},
+			args: args{
+				pod: newGuaranteedPodWithUID("fake-uid-1"),
+			},
+			wantCgroupName:      NewCgroupName(qosContainersInfo.Guaranteed, "podfake-uid-1"),
+			wantLiteralCgroupfs: NewCgroupName(qosContainersInfo.Guaranteed, "podfake-uid-1").ToCgroupfs(),
+		}, {
+			name: "pod with qos guaranteed and systemd",
+			fields: fields{
+				cgroupManager: NewCgroupManager(nil, "systemd"),
+			},
+			args: args{
+				pod: newGuaranteedPodWithUID("fake-uid-2"),
+			},
+			wantCgroupName:      NewCgroupName(qosContainersInfo.Guaranteed, "podfake-uid-2"),
+			wantLiteralCgroupfs: NewCgroupName(qosContainersInfo.Guaranteed, "podfake-uid-2").ToSystemd(),
+		}, {
+			name: "pod with qos burstable and cgroupfs",
+			fields: fields{
+				cgroupManager: NewCgroupManager(nil, "cgroupfs"),
+			},
+			args: args{
+				pod: newBurstablePodWithUID("fake-uid-3"),
+			},
+			wantCgroupName:      NewCgroupName(qosContainersInfo.Burstable, "podfake-uid-3"),
+			wantLiteralCgroupfs: NewCgroupName(qosContainersInfo.Burstable, "podfake-uid-3").ToCgroupfs(),
+		}, {
+			name: "pod with qos burstable and systemd",
+			fields: fields{
+				cgroupManager: NewCgroupManager(nil, "systemd"),
+			},
+			args: args{
+				pod: newBurstablePodWithUID("fake-uid-4"),
+			},
+			wantCgroupName:      NewCgroupName(qosContainersInfo.Burstable, "podfake-uid-4"),
+			wantLiteralCgroupfs: NewCgroupName(qosContainersInfo.Burstable, "podfake-uid-4").ToSystemd(),
+		}, {
+			name: "pod with qos best-effort and cgroupfs",
+			fields: fields{
+				cgroupManager: NewCgroupManager(nil, "cgroupfs"),
+			},
+			args: args{
+				pod: newBestEffortPodWithUID("fake-uid-5"),
+			},
+			wantCgroupName:      NewCgroupName(qosContainersInfo.BestEffort, "podfake-uid-5"),
+			wantLiteralCgroupfs: NewCgroupName(qosContainersInfo.BestEffort, "podfake-uid-5").ToCgroupfs(),
+		}, {
+			name: "pod with qos best-effort and systemd",
+			fields: fields{
+				cgroupManager: NewCgroupManager(nil, "systemd"),
+			},
+			args: args{
+				pod: newBestEffortPodWithUID("fake-uid-6"),
+			},
+			wantCgroupName:      NewCgroupName(qosContainersInfo.BestEffort, "podfake-uid-6"),
+			wantLiteralCgroupfs: NewCgroupName(qosContainersInfo.BestEffort, "podfake-uid-6").ToSystemd(),
+		},
+	}
+
+	for _, tt := range tests {
+		pcm := &podContainerManagerImpl{
+			cgroupManager:     tt.fields.cgroupManager,
+			qosContainersInfo: qosContainersInfo,
+		}
+
+		t.Run(tt.name, func(t *testing.T) {
+			actualCgroupName, actualLiteralCgroupfs := pcm.GetPodContainerName(tt.args.pod)
+			assert.Equalf(t, tt.wantCgroupName, actualCgroupName, "Unexpected cgroup name for pod with UID %s, container resources: %v", tt.args.pod.UID, tt.args.pod.Spec.Containers[0].Resources)
+			assert.Equalf(t, tt.wantLiteralCgroupfs, actualLiteralCgroupfs, "Unexpected literal cgroupfs for pod with UID %s, container resources: %v", tt.args.pod.UID, tt.args.pod.Spec.Containers[0].Resources)
```

Comment:
```suggestion
			require.Equalf(t, tt.wantCgroupName, actualCgroupName, "Unexpected cgroup name for pod with UID %s, container resources: %v", tt.args.pod.UID, tt.args.pod.Spec.Containers[0].Resources)
			require.Equalf(t, tt.wantLiteralCgroupfs, actualLiteralCgroupfs, "Unexpected literal cgroupfs for pod with UID %s, container resources: %v", tt.args.pod.UID, tt.args.pod.Spec.Containers[0].Resources)
```

prefer require over assert - it will terminate tests rather than stacking errors and making them harder to reason about.

Reply:
Thanks for your suggestions! Updated!